### DISCUSSION TITLE
Some more MoonRay in Houdini questions

### DISCUSSION THREAD

**Discussion by BrianHanke** [dreamworksanimation/openmoonray#84] [Q&A] [answered]
Hi all, I found some more things I'm not sure about, wondering if these are bugs or missing features:

1. I notice that the hd_usd2rdl command seems to mess up a lot of scene elements when processing a USD saved from Solaris. For example, in one test scene I made it adds a single-sided geometry flag to a reference Alembic that ruins glass and SSS materials. It also doesn't pick up max samples, among other parameters.
2. All the MoonRay tabs in Render Geometry Settings nodes are blank. There are named tabs but no parameters. Normally you'd expect to be able set subdivision, visibility and other stuff this way.
3. Trying to render using a USD Render ROP just saves a black image to disk. It shows activity in the Render Scheduler and takes the amount of time you'd expect, but no picture... Same with rendering to MPlay.

**Comment by golfingmateo** [comment]
1.  Did you set the sidedness of the Geometry?   I believe if it is not defined, Alembic geometry defaults to single sided. [USD Link](https://openusd.org/release/api/class_usd_geom_gprim.html#a63367a54086e014dda98cc79ad5f8602) .   I think the code where this happens is here: [Moonray Geometry Link](https://github.com/dreamworksanimation/hdMoonray/blob/8fc04a550a80e819cc17f48dccced7a976ab1ee7/src/Geometry.cc#L386)
2. I'll have to test this out.  Internally I am seeing all of the parms but there may be something bad with our public ds files
3. Is hdMoonray working in the viewport?

**Comment by BrianHanke** [comment]
@golfingmateo Thanks for the reply!

1. Yes, that fixed it. Setting doubleSided to true in an Edit LOP made for a correct RDLA conversion.
2. Here's what I see on any Render Geometry Settings node:
![moonray](https://user-images.githubusercontent.com/59420805/229580200-b78de7aa-02b5-4b43-816d-61cf1b38328a.png)
3. Yep, MoonRay is working in the viewport and I can export the scene to USD, then RDLA/B and it renders fine there too. It's just in the USD Render ROP and MPlay that it fails. I tested the same scene in Karma and it renders to disk no problem. Here it took 39 seconds to render with MoonRay, which is correct, but the EXR is black.
![moonray2](https://user-images.githubusercontent.com/59420805/229581516-0b02cdf0-f264-4d10-872f-ea2fb309910c.png)

**Reply by cherrychen0602** [reply]
> 3. Yep, MoonRay is working in the viewport and I can export the scene to USD, then RDLA/B and it renders fine there too. It's just in the USD Render ROP and MPlay that it fails. I tested the same scene in Karma and it renders to disk no problem. Here it took 39 seconds to render with MoonRay, which is correct, but the EXR is black.

The render is black because husk is not calling Resolve before saving the image. I reported this issue to sidefx and they acknowledged this was a bug and said it had been fixed in Houdini daily builds 19.0.924 and 19.5.557.  I was told the HoudiniUsdBridge would be updated soon.

**Reply by PatrickZeller** [reply]
Hey all.

i guess this file is incomplete houdini/soho/parameters/moonray_Geometry.ds

@BrianHanke thanks btw for the houdini moonray build guide helped a lot!

**Reply by golfingmateo** [reply]
You are correct @PatrickZeller .  That file is empty.  Thanks.  We will fix that

**Comment by BrianHanke** [comment]
@PatrickZeller Cool, glad it helped!

**Comment by BrianHanke** [comment]
Any insights on the USD Render ROP issue? It's a bit inconvenient to not be able to render to disk directly from Houdini.

**Reply by golfingmateo** [reply]
There was a bug with Houdini. So it only works with versions later than 19.0.924/19.5.557. 
It also requires a pretty exact setup of RenderVariables.

**Reply by BrianHanke** [reply]
> It also requires a pretty exact setup of RenderVariables.

Thanks. Can you be more specific about this?

**Reply by cherrychen0602** [reply, marked as answer]
When you set up the RenderVar prim, the name needs to be exactly "beauty" for beauty render. Here is the list of pre-defined aovs that hdMoonray currently support.   https://github.com/dreamworksanimation/hdMoonray/blob/8fc04a550a80e819cc17f48dccced7a976ab1ee7/src/RenderBuffer.cc#L66
Also, you might want to make sure the value of the attribute "driver:parameters:aov:format" on RenderVar prim is correct for your aov. 

Hope it helps!

**Comment by BrianHanke** [comment]
@cherrychen0602 Great, thanks! I'll test it soon.
